libfreerdp/core/listener.c

```diff
@@ -347,8 +352,10 @@ static BOOL freerdp_listener_check_fds(freerdp_listener* instance)
 
 	for (i = 0; i < listener->num_sockfds; i++)
 	{
+		WSAResetEvent(listener->events[i]);
```

Comment:
You probably need to keep my '#ifdef _WIN32' here. Reset event on posix platform (Linux, Mac) is not correct. ResetEvent on Linux would break the read buffer because the socket event is the fd itself. We only need it on Windows

 I agree that it's every tricky for us to understand the difference on socket event between windows and posix platform due to the different implementation.